### PR TITLE
[v16] fix: Address flakiness and revisit TestTeleportClient_Login_local

### PR DESCRIPTION
**Pull request by codingllama**
Backport #47406 to branch/v16.

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -495,6 +495,14 @@ type Config struct {
 
 	// SSHDialTimeout is the timeout value that should be used for SSH connections.
 	SSHDialTimeout time.Duration
+
```
Comment (codingllama) [inline lib/client/api.go:498]: We got a conflict here due to some lines that only exist on master. An easy fix.

**Review by GavinFrazar — APPROVED** [review]

**Comment by codingllama** [comment]
The FTD is timing out, but I don't want to make a bunch (more) changes only in the release branch, especially as it seems OK on master. @zmb3, could I get an exemption?

**Comment by codingllama** [comment]
Friendly ping @zmb3 ?

**Comment by codingllama** [comment]
Friendly ping, @zmb3? Could I get an FTD exemption for the timeouts (it isn't flaking anymore).

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**